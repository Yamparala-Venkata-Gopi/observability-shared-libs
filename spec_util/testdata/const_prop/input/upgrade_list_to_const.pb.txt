# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    list_template: {
      # Both elems refer to constant -- the whole list should be upgraded to
      # a constant.
      elem_templates: {
        ref: {
          method_index: 0
          arg_ref: {
            key: "primitive_constant_arg"
            data_ref: {
              primitive_ref: {}
            }
          }
        }
      }
      elem_templates: {
        ref: {
          method_index: 0
          arg_ref: {
            key: "struct_constant_arg"
            data_ref: {
              struct_ref: {
                full_struct: {}
              }
            }
          }
        }
      }
    }
  }
}

arg_templates: {
  key: "arg2"
  value: {
    list_template: {
      # All elems are constant already -- the whole list should be upgraded to
      # a constant.
      elem_templates: {
        value: {
          primitive: {
            int64_value: {
              value: 522
            }
          }
        }
      }
      elem_templates: {
        value: {
          primitive: {
            int32_value: {
              value: 1203
            }
          }
        }
      }
    }
  }
}
