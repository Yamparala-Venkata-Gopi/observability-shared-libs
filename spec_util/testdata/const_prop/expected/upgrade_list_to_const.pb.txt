# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    value: {
      list: {
        elems: {
          primitive: {
            type_hint: "string_constant_1"
            string_value: {
              value: "7c73565e-0d62-49b0-84e7-3063f4d53d9c"
            }
          }
        }
        elems: {
          struct: {
            fields: {
              key: "foo"
              value: {
                primitive: {
                  type_hint: "string_constant_2"
                  string_value: {
                    value: "1aa4e597-a29d-42d4-8134-3e4ee73aa7a1"
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}

arg_templates: {
  key: "arg2"
  value: {
    value: {
      list: {
        elems: {
          primitive: {
            int64_value: {
              value: 522
            }
          }
        }
        elems: {
          primitive: {
            int32_value: {
              value: 1203
            }
          }
        }
      }
    }
  }
}
